File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ret = 0;
        int firstyBuy = prices[0];

        for (auto& price : prices){
            if (price <= firstyBuy){
                firstyBuy = price;
                continue;
            }
            else{
                ret = max(ret, price - firstyBuy);
            }
        }
        return ret;
    }
};

File: Data Structures & Algorithms/partition-labels/submission-1.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        charLastIdx = {}
        for i, v in enumerate(s):
            charLastIdx[v] = i

        ret = []
        size, end = 0, 0
        for i, v in enumerate(s):
            size += 1
            end = max(end, charLastIdx[v])

            if end == i:
                ret.append(size)
                size = 0
        
        return ret
                